photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/DataProcessingUtils.scala

```diff
@@ -117,9 +117,24 @@ object DataProcessingUtils {
     } else {
       1.0
     }
+
+    val metaMap = Utils.getMapAvro(record, AvroFieldNames.META_DATA_MAP)
```

Comment:
Some folks may argue that ```Null``` is evil and we should not see any ```Null``` in Scala (e.g., an empty ```Map``` here instead). But this is a minor issue and I am fine with the nullable way of handling things.